### PR TITLE
Move domain_certfile to the 'general' section

### PR DESCRIPTION
**Pull request by chrzaszcz**
Motivation:
- This option affects s2s and c2s.
- All s2s options will be placed in a map and we want `domain_certfile` outside of it.

**Notes**
This option could be changed so that each `certfile` is specified in the relevant `host_config` section, but:
- There should be no global value - extra logic would be needed for this.
- It would be a bit more difficult migration.
- All certificate files are in one place, which looks tidy.

This is why I decided to keep it as a map.

The `domain` can be a host type, but only for c2s, so I decided to keep the option naming unchanged for now.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_23](https://circleci.com/gh/esl/MongooseIM/102507) / small_tests / 6626ebf9537a42f107cd997401c61cbce45edead
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3512/102507/small_tests.23.1) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3512/102507/small_tests.23.1/small/ct_run.mongooseim@localhost.2022-01-25_08.14.21/lib.mongooseim.logs/run.2022-01-25_08.14.22/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/102536) / small_tests / ecc71977a6f0bd8ae610f33c28a0b2c3eff3cbab
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3512/102536/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3512/102536/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-01-25_08.23.30/lib.mongooseim.logs/run.2022-01-25_08.23.31/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_23](https://circleci.com/gh/esl/MongooseIM/102579) / small_tests / 0a3897a77d2339e5ffcbe3a46d91d5590d5d8f81
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3512/102579/small_tests.23.1) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3512/102579/small_tests.23.1/small/ct_run.mongooseim@localhost.2022-01-25_08.27.24/lib.mongooseim.logs/run.2022-01-25_08.27.24/suite.log.html)

File: doc/configuration/general.md [hunk 1]
```diff
@@ -220,3 +220,25 @@ Replaces [Cowboy](https://github.com/ninenines/cowboy)'s default name returned i
 
 When specified, will terminate certain processes (e.g. client handlers) that have more messages accumulated in the queue than the specified limit, to prevent resource exhaustion.
 This option is set for C2S, outgoing S2S and component connections and can be overridden for particular `s2s` or `service` listeners in their configurations. **Use with caution!**
+
+### `s2s.domain_certfile`
```
Comment (gustawlippa) [inline doc/configuration/general.md:224]: `general.domain_certfile` here

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/102613) / small_tests / 302b41f3b053546ace8b53ca9348d250ff97a8a7
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3512/102613/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3512/102613/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-01-25_08.56.59/lib.mongooseim.logs/run.2022-01-25_08.56.59/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/102642) / small_tests / 4bd393ad5358f2d83868416b79510284dfd8fd7e
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3512/102642/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3512/102642/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-01-25_09.32.53/lib.mongooseim.logs/run.2022-01-25_09.32.53/suite.log.html)

**Review by gustawlippa — APPROVED** [review]
Looks good, the option does _much_ more sense in the `general` section.

**Pull request merged**